### PR TITLE
Checkout a custom branch of training data repo for model regression tests

### LINKED ISSUE #7357: Checkout a custom branch of training data repo for model regression tests
**Description of Problem**:
Currently, the model regression tests checks out the `master` branch of the training data [repo](https://github.com/RasaHQ/training-data/). This is sometimes limiting if we want to try running a custom configuration on a dataset as that custom configuration is not present on the `master` branch and we wouldn't want to place it there as well because the scheduled model regression tests uses that branch.

**Overview of the Solution**:
It looks like the repository is checked out [here](https://github.com/RasaHQ/training-data/). Adding an extra step where a custom branch of the training data repo is checked out and users can change it in their PR(temporarily) will allow for this.

### PR DESCRIPTION
**Pull request by tczekajlo**
**Proposed changes**:
- Resolves #7357 

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by tczekajlo** [comment]
```yml
include:
 - dataset: ["Carbon Bot"]
   config: ["Sparse + DIET(bow) + ResponseSelector(bow)"]

```

**Comment by tczekajlo** [comment]
@dakshvar22 Can you 👀  if you have a minute? 

A branch name can be defined by adding the `branch` field into the configuration, e.g. https://github.com/RasaHQ/rasa/pull/7521#issuecomment-743200032

**Comment by dakshvar22** [comment]
Amazing! Two observations/questions:
1. Can we change the key to `dataset_branch` to be more explicit?
2. Is the key optional, or is it always needed to be defined?

**Comment by tczekajlo** [comment]
> Amazing! Two observations/questions:
> 
> 1. Can we change the key to `dataset_branch` to be more explicit?

Yes, sure, I'll change it.

> 2. Is the key optional, or is it always needed to be defined?

It's optional, if it's not defined the master branch is the default.

File: .github/configs/mr-test-example.yaml [hunk 1]
```diff
@@ -26,6 +26,13 @@
 ##    config: ["all"]
 ##  - dataset: ["Hermit"]
 ##    config: ["Sparse + DIET(seq) + ResponseSelector(t2t)", "BERT + DIET(seq) + ResponseSelector(t2t)"]
+#
+## Example: Define branch to check-out for a dataset repository. Default branch is 'master'
+## branch: "test-branch"
```
Comment (dakshvar22) [inline .github/configs/mr-test-example.yaml:31]:
```suggestion
## dataset_branch: "test-branch"
```

**Review by dakshvar22 — APPROVED** [review]
Functionality LGTM! I'll test it by pushing a small change to one of the configs in a custom branch. 👍

**Comment by dakshvar22** [comment]
@tczekajlo Results look good to me 👍

**Pull request merged**